admin/refresh_spdx/add_licenses.py

```diff
@@ -56,38 +101,61 @@ def gather_target_names(text):
 """
 
 
-def add_new(already_declared, licenses):
+def insert_new(already_declared, licenses):
   template = Template((
-      '\nlicense_kind(\n'
+      'license_kind(\n'
       '    name = "${licenseId}",\n'
       '    conditions = [],\n'
       '    url = "https://spdx.org/licenses/${licenseId}.html",\n'
-      ')\n'))
+      ')'))
 
-  ret = ''
+  ret = []
   for license in licenses:
-    if license['licenseId'] in already_declared:
-      continue
-    ret += template.substitute(license)
-  return ret
+    id = license['licenseId']
+    old = already_declared.get(id)
+    if not old:
+      print('Adding:', id)
+      already_declared[id] = template.substitute(license)
 
 
 def update_spdx_build():
   """Update //licenses/spdx/BUILD with new license kinds."""
 
-  build_path = '../../licenses/spdx/BUILD'
+  license_json = load_json('licenses.json')
+
+  # Find workspace root
+  build_path = 'licenses/spdx/BUILD'
+  found_spdx_build = False
+  for i in range(5):  # Simple regression failure limiter
+    if os.access('WORKSPACE', os.R_OK) and os.access(build_path, os.R_OK):
+      found_spdx_build = True
+      break
+    os.chdir('..')
+  if not found_spdx_build:
+    print('Could not find', build_path)
+    return 1
+
   with open(build_path, 'r') as fp:
     current_file_content = fp.read()
-  already_declared = gather_target_names(current_file_content)
 
-  license_json = load_json('licenses.json')
-  new_rules = add_new(already_declared, license_json['licenses'])
+  preamble, kinds = parse_build_file(current_file_content)
+  insert_new(kinds, license_json['licenses'])
   with open(build_path, 'w') as fp:
-    fp.write(current_file_content)
-    fp.write(new_rules)
+    fp.write(preamble)
+    for license in sorted(kinds.keys(), key=lambda x: x.lower()):
+      fp.write('\n')
+      fp.write(kinds[license])
+      fp.write('\n')
 
 
 def main():
+  lc_all = os.environ.get('LC_ALL')
+  if lc_all != 'en_US.UTF-8':
+    print('Your environment settings will resort the file badly.')
```

Comment:
Do you mean to use "format" or another word instead of "resort"?

```suggestion
    print('Your environment settings will format the file badly.')
```

Reply:
I meant reorder. Changed.